### PR TITLE
Add fontFamily and fontSize options - Fixes #312

### LINKED ISSUE #312: Expose font-family property for labels / tooltips
Would be great to expose the fontFamily property for the axes and tooltips.

### PR DESCRIPTION
**Pull request by TomBecker-BD**
Provide options for setting the chart fonts. The options apply to the axis and legend. The fonts can be specified directly in the options, or in .axis and .legend styles. See test/visual/test_cases/fonts/ for demos.

### CODE REVIEW

File: src/utils/legend.coffee [hunk 1]
```diff
@@ -107,8 +108,8 @@
             item.append('text')
               .attr(
                 'class': (d, i) -> "legendText series_#{i}"
-                'font-family': 'Courier'
-                'font-size': 10
+                'font-family': options.fontFamily
+                'font-size': options.fontSize
```
Comment (chaosmail) [inline src/utils/legend.coffee]: We are adding loads of stuff here to `attr` that should better go to `style`; could you change this please

**Comment by chaosmail** [comment]
Hi @TomBecker-BD wow, nice work! Usually, it would be better to discuss such features upfront, but this one looks really good.

I have a few remarks:
* the first one, I mentioned directly in the line. we have the problem that we added some properties as attribute whereas they should be added as styles
* there are some errors in the Travis test, it seems there are 2 files missing for the visual tests: `expected.png` and a `desc` file. Could you add them please.
* would be create if you could squash the commits into one if everything is ready
* Could you add `Fixes #312` to the initial description of the PR? like this the issue will be closed automatically once we merge this PR

Thanks for the great work, let me know if you need any help with the above remarks.

Best,
Christoph

**Comment by TomBecker-BD** [comment]
This is my first time doing this so I appreciate the help. I was not able to set up the visual test environment (work dev is on Windows), so I manually created some demo HTML files and stuck them there. Probably I should just remove them.

File: src/utils/legend.coffee (hunk 1 above)
Comment (TomBecker-BD) [inline src/utils/legend.coffee]: OK, good idea.

**Comment by chaosmail** [comment]
@TomBecker-BD no worries, I am glad (and honored) that your first commit will be to our library.

* Don't worry for the visual test for now, I can make a PR on your fork to add the missing files in the end. * Here is a guide on squashing your commits into one: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes

Let me know if you need any more help!

Best,
Christoph

**Comment by hellsan631** [comment]
+1 this. Really looking forward to changing out of the default monospaced text. Would it be possible to have this be defined in CSS instead? Any quick fix is appreciated though.

**Comment by chaosmail** [comment]
I am sorry, I can review the PR the earliest next friday. here is a quick CSS workaround (see issue #312):

```css
.chart *[style] {
  font-size: 11px !important;
  font-family: "Impact", Helvetica, Arial, sans-serif !important;
  -webkit-font-smoothing: antialiased;
}
.chart *[font-family] {
  font-family: "Impact", Helvetica, Arial, sans-serif !important;
}
```

Best,
Christoph

**Comment by hellsan631** [comment]
Ah, thanks so much. Tried a few CSS tricks before and they ended up killing the text so nothing showed up. I might have tried a bit too far down the tree.

**Comment by chaosmail** [comment]
Hi @TomBecker-BD  
I will close this as it is a little bit outdated. I would recommend you to look into the source of the latest version; it is very clean, well organized and written completely in TypeScript.
Best, Christoph

**Pull request closed without merging**